metaflow/plugins/list_cli.py

```diff
@@ -0,0 +1,77 @@
+from metaflow._vendor import click
+from metaflow import Flow
+from metaflow.exception import MetaflowNotFound
+import json
+
+
+@click.group()
+def cli():
+    pass
+
+
+@cli.group(help="List objects pertaining to your flow.")
+def list():
+    pass
+
+
+def _fetch_runs(flow_name, num_runs):
+    counter = 1
+    try:
+        flow = Flow(flow_name)
+    except MetaflowNotFound:
+        flow = None
+    run_list = []
+    if flow:
+        for run in Flow(flow_name).runs():
+            if counter > num_runs:
+                break
+            counter += 1
+            run_list.append(
+                dict(
+                    created=str(run.created_at),
+                    name=flow_name,
+                    id=run.id,
+                    status=run.successful,
+                    finished=run.finished,
+                )
+            )
+    return run_list
+
+
+@click.option(
+    "--num-runs",
+    default=10,
+    type=click.IntRange(1, None),
+    help="Number of runs to show.",
+)
+@click.option(
+    "--as-json",
+    default=False,
+    is_flag=True,
+    help="Print run list as a JSON object",
+)
+@click.option(
+    "--file",
+    default=None,
+    help="Save the run list to file as json.",
+)
+@list.command(help="List recent runs for your flow.")
+@click.pass_context
+def runs(ctx, num_runs, as_json, file):
```

Comment:
Can you also add `--user` and `--my-runs` flags as well? - take a look at `batch_cli.py`

Reply:
:thumbs_up:

Reply:
Thanks I implemented this in https://github.com/Netflix/metaflow/pull/1037/commits/e0682b15a576989cfd408db7708e26870bc282ed